lobster/related.lobster

```diff
@@ -0,0 +1,88 @@
+
+import std
+import dictionary
+import json
+
+let TOPN = 5
+
+struct Post:
+    _id: string
+    title: string
+    tags: [string]
+
+struct RelatedPost:
+    _id: string
+    tags: [string]
+    related: [Post]
+
+def main():
+    let json_content = read_file("../posts.json")
+
+    assert(json_content != nil)
+
+    let posts = JSON.parse<[Post]>(json_content)
+
+    let start = seconds_elapsed()
+
+    let tag_map = dictionary<string, [int]> ( 100 )
```

Comment:
how many tags are you typically expecting? also may want a prime number as the size (see dictionary.lobster)

Reply:
Max 100 for this test.

Reply:
Then I would use a prime number like `251` (or if you want to even sure there's less collisions, `1009`).